applications/audio/broadcast_config_tool/src/main.c

```diff
@@ -1279,6 +1467,93 @@ static int cmd_program_info(const struct shell *shell, size_t argc, char **argv)
 	return 0;
 }
 
+#define FILE_LIST_BUF_SIZE 512
+static int cmd_file_list(const struct shell *shell, size_t argc, char **argv)
+{
+	int ret;
+	char buf[FILE_LIST_BUF_SIZE];
+	size_t buf_size = FILE_LIST_BUF_SIZE;
+	char *dir_path = NULL;
+
+	if (argc > 2) {
+		shell_error(shell, "Usage: bct file list [dir path]");
+		return -EINVAL;
+	}
+
+	if (argc == 2) {
+		dir_path = argv[1];
+	}
+
+	ret = sd_card_list_files(dir_path, buf, &buf_size);
+	if (ret) {
+		shell_error(shell, "List files err: %d", ret);
+		return ret;
+	}
+
+	shell_print(shell, "%s", buf);
+
+	return 0;
+}
+
+static int cmd_file_select(const struct shell *shell, size_t argc, char **argv)
+{
+	int ret;
+	uint8_t big_idx;
+	uint8_t sub_idx;
+	uint8_t bis_idx;
+
+	if (argc < 4) {
+		shell_error(shell,
+			    "Usage: bct file select <file path> <BIG index> <subgroup index> "
+			    "<BIS index> [loop file 1/0]");
+		return -EINVAL;
+	}
+
+	ret = argv_to_indexes(shell, argc, argv, &big_idx, 2, &sub_idx, 3);
+
+	if (!is_number(argv[4])) {
+		shell_error(shell, "BIS index must be a digit");
+		return -EINVAL;
+	}
+
+	bis_idx = (uint8_t)atoi(argv[4]);
+
+	if (bis_idx >= CONFIG_BT_BAP_BROADCAST_SRC_STREAM_COUNT) {
+		shell_error(shell, "BIS index out of range");
```

Comment:
To assist a bit more, we can print
```suggestion
		shell_error(shell, "BIS index out of range %d > %d", bis_idx, CONFIG_BT_BAP_BROADCAST_SRC_STREAM_COUNT - 1);
```